### DISCUSSION TITLE
libpac or other PAC file support?

### DISCUSSION THREAD

**Discussion by ejdarrow** [audacious-media-player/audacious#71] [Q&A]
Hi all. I recognize this is a 'most likely no' question off the bat, but does anyone know if PAC (soundblaster) audio files have a plugin of any sort? I saw in the forums that Audacious is NOT plugin compatible with XMMS, and the last project I saw for interacting with or converting those files is specifically an xmms plugin.

If there is not a known plugin/project associated with this, is the process of converting an xmms plugin to be compatible with Audacious a known procedure, or is it so different now that it's case-by-case no matter what?

**Comment by mschwendt** [comment]
Seems to be one of the old proprietary formats from Creative Labs for Soundblaster Studio II. I wonder, has file format documentation ever been published? Or does the XMMS plugin only rely on an emulator?

With regard to plugins, I would say that writing plugins is easier with Audacious than with XMMS. As an additional benefit, if a plugin wants to offer a preferences/settings GUI, plugin authors don't need to do that with native GTK anymore since Audacious implements an own widget abstraction API for that.

**Reply by wildstar84** [reply]
You are correct in that Audacious uses it's own config widgets, namely in order to work same in both Qt & GTK[2|3], and skins interfaces.  As far as being easier than in XMMS, I couldn't say, having never visited any of their plugins' code, but I suspect the code may be somewhat similar.   Someone might be able to take the code for that plugin and convert it to Aud. format, perhaps?
